packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx

```diff
@@ -113,11 +133,32 @@ export const StyledBaseInput = React.forwardRef<TextInput, StyledBaseInputProps>
       type,
       numberOfLines,
       isTextArea,
+      hasPopup,
       ...props
     },
     ref,
   ) => {
-    return (
+    return hasPopup ? (
+      <StyledNativeBaseButton
+        // the types of styled-components for react-native is creating a mess, so there's no other option but to type `ref` as any
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ref={ref as any}
+        // @ts-expect-error: not able to type onClick for both web and native
```

Comment:
In this file it should be typed only for native right?

Reply:
So we can't type `onClick` because it is coming from common prop so if we type it for native here, the parent component from where we're passing down the prop starts failing for web.


We can type `onPress` to have web's `onClick` but I think it is better to ignore the error than incorrectly overriding the onPress

Reply:
Anyway we're not writing much of a code inside of that prop so don't see much value in having correct type there

Reply:
Update: Changed the implementation a bit to make it similar to our onFocus and onBlur events. That fixed the TS issues - https://github.com/razorpay/blade/pull/932#discussion_r1085081773